setup.py

```diff
@@ -99,10 +99,10 @@ def build_extension_from_pyx(pyx_path, extra_sources_paths=None):
 
 
 # Please see conda/meta.yaml for other binary dependencies
-install_requires = ['numpy>=1.10,<2.0',
-                    'scipy>=0.16,<1.0',
-                    'matplotlib>=1.4,<3.0',
-                    'pillow>=3.0,<5.0']
+install_requires = ['numpy>=1.10',
```

Comment:
Won't this mean eventually menpo will just break due to incompatibilities? Shouldn't we bump the majors as we know if they are actually compatible?

Reply:
It's a trade off I guess between:

- being sure that stuff will work
- integrating easily to existing environments
- the maintenance burden we want to take on testing stuff

As of now, latest versions of our deps are:

numpy: 1.15
scipy: 1.10
matplotlib: 2.2.2
pillow: 5.2.0

so we're blocking user's from installing menpo with latest versions of 2 out of our 4 deps right now. As those deps are the core of the scientific python stack, I don't think user's will downgrade to install menpo, they'll just not install menpo.

If we were working on menpo every day, having a < limit is definitely preferable - we'd quickly test new major revisions as a matter of course and bump aggressively. Now we are more in a maintenance mode, I'm not sure if forcing a < is a good trade off.

Yeah this would mean menpo would run into (likely minor) incompatibilities in time, but we would likely hear from our users on the mailing list/issues when that happens, and we can then accept PRs and quickly issue patches. In the alternative where we keep pinning, users will never hit those minor issues due to the pinning (great), but will I think be put off when they are told menpo won't install with scipy 1.10. I don't know if we have the bandwidth to be proactive in testing + bumping our deps to avoid that going forward (we've not done too well the last 12 months).

Btw we can leave conda with pinned deps for now, so this is only for the pip install case in my mind.

Reply:
Fair - thought the burden might be similar in the end as at some point someone has to go in and fix the errors and re-release